eval.py

```diff
@@ -2,47 +2,108 @@
 import fire
 import numpy as np
 import yaml
-from selection.serialization import deserialize
+from pathlib import Path
+import pandas as pd
 
+#Constants
+label_idx = 0 #IDs look like "en/clips/and/common_voice..." this is the index of the label
 
-def main(eval_file, input_samples, train_file="train.yaml"):
 
-    eval_data = deserialize.deserialize_from_pb(eval_file)
-    eval_target = eval_data["target_mswc_vectors"]
-    eval_nontarget = eval_data["nontarget_mswc_vectors"]
-    eval_x = np.vstack([eval_target, eval_nontarget])
-    eval_y = np.concatenate(
-        [np.ones(eval_target.shape[0]), np.zeros(eval_nontarget.shape[0])]
-    )
+def create_train_sets(selected_data, allowed_training_embeddings, training_classes, train_dataset):
+    train_x = [] #embeddings
+    train_y = [] #numerical index of training_classes 
+    for selected_id in selected_data["nontargets"]:
+        if not selected_id in allowed_training_embeddings['nontargets']:
+            print("invaid embedding ID: ", selected_id)
+            raise ValueError('Selected Embedding ID not present in allowed training data')
 
-    sample_data = deserialize.deserialize_from_pb(input_samples)
-    target_vectors = sample_data["target_mswc_vectors"]
-    target_ids = sample_data["target_ids"]
-    nontarget_vectors = sample_data["nontarget_mswc_vectors"]
-    nontarget_ids = sample_data["nontarget_ids"]
+        label = Path(selected_id).parts[label_idx]
```

Comment:
i'm confused - if idx is 0, doesn't it return `en`? (in `en/clips/and/common_voice_12345.opus`)

Reply:
the IDs in the parquet files are actually of the form `left/common_voice_en_2145.wav`

I made it a variable cause I thought that might change

Reply:
oh right! Good catch lol